crates/forge/tests/integration/trace.rs

```diff
@@ -5,66 +5,129 @@ use test_utils::running_tests::run_test_case;
-                assert_eq!(get_last_call_trace().len(), 0);
-            
                 let checker = declare('TraceInfoChecker');
-                assert_eq!(get_last_call_trace().len(), 0);
             
                 let checker_address = checker.deploy(@array![]).unwrap();
-                assert_eq!(get_last_call_trace().len(), 0); // no constructor
             
-                let proxy_address = proxy.deploy(@array![checker_address.into()]).unwrap();
-                assert_trace_after_proxy_deploy(get_last_call_trace(), proxy_address, checker_address);
+                let proxy_address1 = proxy.deploy(@array![checker_address.into()]).unwrap();
             
-                let (proxy_address_2, _) = deploy_syscall(
+                let (proxy_address2, _) = deploy_syscall(
                     proxy.class_hash, 0, array![checker_address.into()].span(), false
                 )
                     .unwrap_syscall();
-                assert_trace_after_proxy_deploy(get_last_call_trace(), proxy_address_2, checker_address);
             
                 let proxy_address_3 = proxy
                     .deploy_at(@array![checker_address.into()], 123.try_into().unwrap())
                     .unwrap();
-                assert_trace_after_proxy_deploy(get_last_call_trace(), proxy_address_3, checker_address);
+            
+                assert_trace(
+                    get_call_trace(), proxy_address1, proxy_address2, proxy_address_3, checker_address
+                );
             }
             
-            fn assert_trace_after_proxy_deploy(
-                trace: Array<CallEntryPoint>, proxy_address: ContractAddress, checker_address: ContractAddress
+            fn assert_trace(
+                trace: CallTrace,
+                proxy_address1: ContractAddress,
+                proxy_address2: ContractAddress,
+                proxy_address3: ContractAddress,
+                checker_address: ContractAddress
             ) {
-                let expected_trace = array![
-                    CallEntryPoint {
-                        entry_point_type: EntryPointType::Constructor,
-                        entry_point_selector: selector!("constructor"),
-                        calldata: array![checker_address.into()],
-                        contract_address: proxy_address,
-                        caller_address: test_address(),
-                        call_type: CallType::Call,
-                    },
-                    CallEntryPoint {
+                let expected_trace = CallTrace {
+                    entry_point: CallEntryPoint {
                         entry_point_type: EntryPointType::External,
-                        entry_point_selector: selector!("from_proxy"),
-                        calldata: array![1],
-                        contract_address: checker_address,
-                        caller_address: proxy_address,
+                        entry_point_selector: selector!("TEST_CONTRACT_SELECTOR"),
```

Comment:
Isn't the top-most call an implementation detail and shouldn't it be abstracted away from the user?